boot/nxboot/loader/boot.c

```diff
@@ -403,6 +440,42 @@ static int perform_update(struct nxboot_state *state, bool check_only)
   return OK;
 }
 
+#ifdef CONFIG_NXBOOT_COPY_TO_RAM
+int nxboot_ramcopy(void)
+{
+  int primary;
+  struct nxboot_img_header header;
+  ssize_t bytes;
+  static uint8_t *buf;
+  size_t size;
+
+  primary = flash_partition_open(CONFIG_NXBOOT_PRIMARY_SLOT_PATH);
+  if (primary < 0)
+    {
+      return ERROR;
+    }
+
+  get_image_header(primary, &header);
+  buf = malloc(header.size);
+  if (!buf)
+    {
+      return ERROR;
```

Comment:
Close primary on error. Same after `pread`.

Reply:
As above, but agreed and will do it for thoroughness and to cover all use cases :-)

Reply:
done